src/components/FlowRunTimeline.vue

```diff
@@ -236,32 +243,128 @@
+    const isValueVisible = expandedSubFlowRuns.value.has(id)
+
+    if (isValueVisible) {
+      const subFlowRun = expandedSubFlowRuns.value.get(id)
+      subFlowRun?.subscription.unsubscribe()
+      expandedSubFlowRuns.value.delete(id)
+      return
+    }
+
+    const subscription = useSubscription(
+      api.flowRuns.getFlowRunsTimeline,
+      [id],
+      { interval: 5000 },
+    )
+
+    const data = computed(() => {
+      return removeNullStartsAndSort(subscription.response ?? [])
+    })
+
+    expandedSubFlowRuns.value.set(id, {
+      data,
+      subscription,
+    })
+  }
+
+  function removeNullStartsAndSort(nodes: TimelineNode[]): GraphTimelineNode[] {
+    return nodes.filter(isValidGraphTimelineNode)
+      .sort((nodeA, nodeB) => {
+        return nodeA.start.getTime() - nodeB.start.getTime()
+      })
+  }
+
+  const rootSubFlowRunIds = computed<string[]>(() => {
+    return graphData.value
+      .filter(hasSubFlowRunId)
+      .map((node) => node.subFlowRunId)
+  })
+  const expandedSubFlowRunIds = computed<string[]>(() => {
+    return Array.from(expandedSubFlowRuns.value.values())
+      .flatMap(subFlowRun => 'value' in subFlowRun.data ? subFlowRun.data.value : subFlowRun.data)
+      .filter(hasSubFlowRunId)
+      .map((node) => node.subFlowRunId)
+  })
+  const allSubFlowRunIds = computed<string[]>(() => {
+    return [...rootSubFlowRunIds.value, ...expandedSubFlowRunIds.value]
+  })
+
+  const subFlowRunsSubscriptionFilter = computed<Parameters<typeof api.flowRuns.getFlowRuns> | null>(() => {
+    if (allSubFlowRunIds.value.length < 1) {
+      return null
+    }
+
+    return [
+      {
+        flowRuns: {
+          id: allSubFlowRunIds.value,
+        },
+      },
+    ]
+  })
+  const subFlowRunsSubscription = useSubscriptionWithDependencies(
+    api.flowRuns.getFlowRuns,
+    subFlowRunsSubscriptionFilter,
+  )
+  const subFlowRunLabels = computed(() => {
+    return (subFlowRunsSubscription.response ?? [])
+      .reduce((acc, curr) => {
+        if (curr.name) {
+          acc.set(curr.id, curr.name)
+        }
+        return acc
+      }, new Map<string, string>())
+  })
 
   function getStateColor(cssVariable: string): string {
     return bodyStyles.getPropertyValue(cssVariable).trim()
   }
 
-  const stateColors: Record<string, string> = {
-    completed: getStateColor('--state-completed-600'),
-    running: getStateColor('--state-running-600'),
-    scheduled: getStateColor('--state-scheduled-600'),
-    pending: getStateColor('--state-pending-600'),
-    failed: getStateColor('--state-failed-600'),
-    cancelled: getStateColor('--state-cancelled-600'),
-    crashed: getStateColor('--state-crashed-600'),
-    paused: getStateColor('--state-paused-600'),
+  const stateColors: Record<string, Record<'default' | 'hover', string>> = {
+    completed: {
+      default: getStateColor('--state-completed-600'),
+      hover: getStateColor('--state-completed-700'),
+    },
```

Comment:
Seems like there's no need to write all of these out - can you map over one of our states arrays instead?

Reply:
I haven't tested every color, I imagine something like a yellow might be tweeked at some point? Idk what the value of a map would really give here over a simple object?

Reply:
ahh I see.. For some reason I couldn't find a state array before, but i see now: https://github.com/PrefectHQ/prefect-ui-library/blob/d055c46f25e52bf8a5fd1fffd874080675b9aacb/src/types/states.ts#L3